### PR TITLE
Display last GitHub status message if cloning fails

### PR DESCRIPTION
**Pull request by BanzaiMan**
Resolves https://github.com/travis-ci/travis-ci/issues/1430.

### CODE REVIEW

**Comment by BanzaiMan** [comment]
The status message would look like this:

```sh-session
$ curl -sL https://status.github.com/api/last-message.json | jq -r .[]
good
Everything operating normally.
2017-01-13T16:55:51Z
```

File: lib/travis/build/git/clone.rb [hunk 1]
```diff
@@ -63,6 +71,16 @@ def dir
           def config
             data.config
           end
+
+          def github?
+            if md = /[^@]+@(.*):/.match(data.source_url)
```
Comment (meatballhat) [inline lib/travis/build/git/clone.rb]: It's a nitpick, but assigning in an `if` statement is definitely a no-no according to the ruby style guide.

File: lib/travis/build/git/clone.rb [hunk 2]
```diff
@@ -63,6 +71,16 @@ def dir
...
+              # we will assume that the URL looks like one for git+ssh; e.g., git@github.com:travis-ci/travis-build.git
+              host = md[1]
```
Comment (meatballhat) [inline lib/travis/build/git/clone.rb]: Should this `host` assignment also have `.downcase`?

File: lib/travis/build/git/clone.rb [hunk 3]
```diff
@@ -63,6 +71,17 @@ def dir
...
+
+          def github?
+            md = /[^@]+@(.*):/.match(data.source_url)
+            if md
+              # we will assume that the URL looks like one for git+ssh; e.g., git@github.com:travis-ci/travis-build.git
+              host = md[1]
+            else
+              host = URI.parse(data.source_url).host
+            end
+            host.downcase == 'github.com' || host.end_with?('.github.com')
```
Comment (meatballhat) [inline lib/travis/build/git/clone.rb]: Shouldn't the right side of the `||` be `.downcase`'d, too?

**Pull request merged**